lib/ansible/executor/playbook_executor.py

```diff
@@ -168,7 +168,7 @@ def run(self):
                 # send the stats callback for this playbook
                 if self._tqm is not None:
                     if C.RETRY_FILES_ENABLED:
-                        retries = list(set(self._tqm._failed_hosts.keys() + self._tqm._unreachable_hosts.keys()))
+                        retries = list(set(list(self._tqm._failed_hosts.keys()) + list(self._tqm._unreachable_hosts.keys())))
```

Comment:
Better to write this like this:
``` python
import itertools
[...]
list(frozenset(itertools.chain(a.keys(), b.keys())))

```
Or if you want to write it out:
``` python
retries = set(a.keys())
retries.update(b.keys())
retries = list(retries)
```

Either of those avoid instantiating two extra lists.